### PR TITLE
Auto-check eggs for update

### PR DESCRIPTION
**Pull request by Boy132**
Adds a new command to check installed eggs for updates. This just fetches the json from the `update_url` and compares it with the current json. (or rather the hashes of the json files) It runs automatically every hour.

If an update is found it will also display a button on the eggs list that will simply re-import the updated egg json.

_This will also mean that if you edit an egg it will detect it as "outdated"!_

### CODE REVIEW

File: app/Console/Commands/Egg/CheckEggUpdates.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace App\Console\Commands\Egg;
+
+use App\Models\Egg;
+use App\Services\Eggs\Sharing\EggExporterService;
+use Exception;
+use Illuminate\Console\Command;
+
+class CheckEggUpdates extends Command
```
Comment (RMartinOscar) [inline app/Console/Commands/Egg/CheckEggUpdates.php:10]:
```suggestion
class CheckEggUpdatesCommand extends Command
```

File: app/Console/Kernel.php [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 namespace App\Console;
 
+use App\Console\Commands\Egg\CheckEggUpdates;
```
Comment (RMartinOscar) [inline app/Console/Kernel.php:5]:
```suggestion
use App\Console\Commands\Egg\CheckEggUpdatesCommand;
```

File: app/Console/Kernel.php [hunk 3]
```diff
@@ -35,6 +36,7 @@ protected function schedule(Schedule $schedule): void
 
         $schedule->command(CleanServiceBackupFilesCommand::class)->daily();
         $schedule->command(PruneImagesCommand::class)->daily();
+        $schedule->command(CheckEggUpdates::class)->hourly();
```
Comment (RMartinOscar) [inline app/Console/Kernel.php:39]:
```suggestion
        $schedule->command(CheckEggUpdatesCommand::class)->hourly();
```

**Review by RMartinOscar — COMMENTED** [review]
`CheckEggUpdates.php` should be `CheckEggUpdatesCommand.php`

File: app/Console/Commands/Egg/CheckEggUpdates.php [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                if (is_null($egg->update_url)) {
+                    $this->comment("{$egg->name}: Skipping (no update url set)");
+
+                    continue;
+                }
+
+                $currentJson = json_decode($exporterService->handle($egg->id));
+                unset($currentJson->exported_at);
+
+                $updatedJson = json_decode(file_get_contents($egg->update_url));
```
Comment (RMartinOscar) [inline app/Console/Commands/Egg/CheckEggUpdates.php:31]: Shouldn't use `file_get_contents`

Comment (Boy132) [reply app/Console/Commands/Egg/CheckEggUpdates.php:31]: `file_get_contents` is fine, we also use it in the egg importer.

**Review by lancepioch — APPROVED** [review]

**Pull request merged**